### PR TITLE
Enable console to set expired-object-all-versions

### PR DESCRIPTION
**Pull request by shtripat**
This PR enables console to set bucket ILM rule for deleting all versions of expired object.

Steps to test:
Create a bucket and enable versioning for the same.
Try to create ILM rule and select option for delete all versions.
Make sure listing of rules works fine.

### CODE REVIEW

**Comment by prakashsvmx** [comment]
@shtripat , please resolve the conflicts

**Comment by shtripat** [comment]
> @shtripat , please resolve the conflicts

Done!

**Comment by harshavardhana** [comment]
Some permission tests failing

```
Run npx testcafe "chrome:headless" web-app/tests/permissions-B/ --skip-js-errors -c 3
 Running tests in:
 - Chrome 121.0.6167.160 / Ubuntu 22.0[4](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:5)
 - Chrome 121.0.6167.160 / Ubuntu 22.04
 - Chrome 121.0.6167.160 / Ubuntu 22.04

 For user with Bucket Write to specific prefix permissions
 ✖ Upload File button is disable and Upload Folder button is enabled on bucket root path

   1) Cannot obtain information about the node because the specified selector does not match any node in the DOM tree.

         | Selector('li')
       > |   .withText('Upload File')

      Browser: Chrome 121.0.6167.160 / Ubuntu 22.04

         30 |      await t
         31 |        .useRole(roles.bucketWritePrefixOnly)
         32 |        .navigateTo("http://localhost:9090/browser/testcafe")
         33 |        .click(uploadButton)
         34 |        .expect(Selector("li").withText("Upload File").hasClass("disabled"))
       > 3[5](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:6) |        .ok()
         3[6](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:7) |        .expect(Selector("li").withText("Upload Folder").hasClass("disabled"))
         3[7](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:8) |        .notOk();
         3[8](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:9) |    },
         3[9](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:10) |  )
         40 |  .after(async (t) => {});

         at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:35:[10](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:11))
         at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:23:71)
         at __awaiter (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:19:[12](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:13))
         at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:28:17)


 ✖ Upload File and Folder buttons are enabled on bucket prefix path

   1) Cannot obtain information about the node because the specified selector does not match any node in the DOM tree.

         | Selector('li')
       > |   .withText('Upload File')

      Browser: Chrome 121.0.6[16](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:17)7.160 / Ubuntu 22.04

         47 |      await t
         48 |        .useRole(roles.bucketWritePrefixOnly)
         49 |        .navigateTo("http://localhost:9090/browser/testcafe/d3JpdGU=")
         50 |        .click(uploadButton)
         51 |        .expect(Selector("li").withText("Upload File").hasClass("disabled"))
       > 52 |        .notOk()
         53 |        .expect(Selector("li").withText("Upload Folder").hasClass("disabled"))
         54 |        .notOk();
         55 |    },
         56 |  )
         57 |  .after(async (t) => {});

         at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:52:10)
         at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:23:71)
         at __awaiter (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:19:12)
         at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:45:[17](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:18))



 2/2 failed ([18](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:19)s)
Error: Process completed with exit code 2.
```

**Comment by shtripat** [comment]
> Some permission tests failing
> 
> ```
> Run npx testcafe "chrome:headless" web-app/tests/permissions-B/ --skip-js-errors -c 3
>  Running tests in:
>  - Chrome 121.0.6167.160 / Ubuntu 22.0[4](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:5)
>  - Chrome 121.0.6167.160 / Ubuntu 22.04
>  - Chrome 121.0.6167.160 / Ubuntu 22.04
> 
>  For user with Bucket Write to specific prefix permissions
>  ✖ Upload File button is disable and Upload Folder button is enabled on bucket root path
> 
>    1) Cannot obtain information about the node because the specified selector does not match any node in the DOM tree.
> 
>          | Selector('li')
>        > |   .withText('Upload File')
> 
>       Browser: Chrome 121.0.6167.160 / Ubuntu 22.04
> 
>          30 |      await t
>          31 |        .useRole(roles.bucketWritePrefixOnly)
>          32 |        .navigateTo("http://localhost:9090/browser/testcafe")
>          33 |        .click(uploadButton)
>          34 |        .expect(Selector("li").withText("Upload File").hasClass("disabled"))
>        > 3[5](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:6) |        .ok()
>          3[6](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:7) |        .expect(Selector("li").withText("Upload Folder").hasClass("disabled"))
>          3[7](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:8) |        .notOk();
>          3[8](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:9) |    },
>          3[9](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:10) |  )
>          40 |  .after(async (t) => {});
> 
>          at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:35:[10](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:11))
>          at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:23:71)
>          at __awaiter (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:19:[12](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:13))
>          at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:28:17)
> 
> 
>  ✖ Upload File and Folder buttons are enabled on bucket prefix path
> 
>    1) Cannot obtain information about the node because the specified selector does not match any node in the DOM tree.
> 
>          | Selector('li')
>        > |   .withText('Upload File')
> 
>       Browser: Chrome 121.0.6[16](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:17)7.160 / Ubuntu 22.04
> 
>          47 |      await t
>          48 |        .useRole(roles.bucketWritePrefixOnly)
>          49 |        .navigateTo("http://localhost:9090/browser/testcafe/d3JpdGU=")
>          50 |        .click(uploadButton)
>          51 |        .expect(Selector("li").withText("Upload File").hasClass("disabled"))
>        > 52 |        .notOk()
>          53 |        .expect(Selector("li").withText("Upload Folder").hasClass("disabled"))
>          54 |        .notOk();
>          55 |    },
>          56 |  )
>          57 |  .after(async (t) => {});
> 
>          at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:52:10)
>          at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:23:71)
>          at __awaiter (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:19:12)
>          at <anonymous> (/home/runner/work/console/console/web-app/tests/permissions-B/bucketWritePrefixOnly.ts:45:[17](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:18))
> 
> 
> 
>  2/2 failed ([18](https://github.com/minio/console/actions/runs/7906473383/job/21581373151#step:9:19)s)
> Error: Process completed with exit code 2.
> ```

Fixed by https://github.com/minio/console/pull/3236

File: go.mod [hunk 1]
```diff
@@ -158,3 +158,5 @@ require (
 	gopkg.in/ini.v1 v1.67.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace github.com/minio/mc v0.0.0-20240209221824-669cb0a9a475 => github.com/shtripat/mc v0.0.0-20240212061450-03c9dfe77107
```
Comment (bexsoft) [inline go.mod:162]: Please remove replace

**Review by bexsoft — CHANGES_REQUESTED** [review]

**Review by prakashsvmx — APPROVED** [review]
LGTM :+1:

**Review by bexsoft — APPROVED** [review]

**Pull request merged**